ge/org.osate.ge.ba/src/org/osate/ge/ba/util/BehaviorAnnexXtextUtil.java

```diff
@@ -48,14 +48,19 @@ public static String getText(final IXtextDocument xtextDocument, final XtextReso
 	/**
 	 * Find offset of character in string that is not commented out
 	 */
-	public static int findUncommentedChar(final String str, final char delim) {
+	public static int findUncommentedTerminationChar(final String str, final char delim) {
+		final String openBrackets = "{[";
+		final String closeBrackets = "]}";
 		final PeekingIterator<Character> charPeekingIt = Iterators
```

Comment:
This is very inefficient. This involves boxing characters as `Character` and allocating a new list.  Use guava's `Lists.charactersOf(str)` to create a list which is a view of the existing string.  This will still box chars as `Character` but will avoid creating a new list and iterating through every character before your loop starts. It would be more efficient to keep things as char but this will still allow usage of `PeekingIterator`.

Reply:
@RyanMcilnay I believe we had a miscommunication regarding my comment. The problem wasn't with `openBrackets` and `closeBrackets`. Those were fine as they were. The problem was with the collect operation. For example, that line can be replaced with something like `final PeekingIterator<Character> charPeekingIt = Iterators.peekingIterator( Lists.charactersOf(str).iterator());`